DnDCharacterSheet/Converters/CoinConverter.cs

```diff
@@ -0,0 +1,24 @@
+﻿using DTOs;
+using Interfaces;
+using Models;
+
+namespace Converters
+{
+    public class CoinConverter : IConverter<Coin, CoinDTO>
+    {
+        public CoinDTO Convert(Coin source)
+        {
+            return new CoinDTO()
+            {
+                Id = source.Id,
+                Name = source.Name,
+                Initials = source.Initials,
```

Comment:
could it happen Coin to be null? have you setup nullable in the projecT?

Reply:
Not possible, and now using Optional Convert is only called when CurrencyRepository successfully returns a Currency.